From 455792ec2f0163ea511cc0939305d8e7aea326ca Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Tue, 8 Apr 2014 02:46:00 -0400
Subject: [PATCH] allow mount to override contexts

mount fails to override the default contexts while running:
    $ mount ip:/sharedFs /mnt/sharedFs -o context="new_context"

Related avc issues:

type=AVC msg=audit(1396939320.262:108): avc:  denied  { relabelfrom } for  pid=2183 comm="mount.nfs" scontext=root:sysadm_r:mount_t:s0-s15:c0.c1023 tcontext=system_u:object_r:user_home_t:s0 tclass=filesystem
type=AVC msg=audit(1396939321.174:110): avc:  denied  { remove_name } for  pid=2183 comm="mount.nfs" name="utab.IuAaSb" dev="tmpfs" ino=30727 scontext=root:sysadm_r:mount_t:s0-s15:c0.c1023 tcontext=system_u:object_r:mount_var_run_t:s0 tclass=dir
type=AVC msg=audit(1396939321.157:109): avc:  denied  { setattr } for  pid=2183 comm="mount.nfs" name="utab.IuAaSb" dev="tmpfs" ino=30727 scontext=root:sysadm_r:mount_t:s0-s15:c0.c1023 tcontext=root:object_r:mount_var_run_t:s0 tclass=file
type=AVC msg=audit(1396939321.174:110): avc:  denied  { rename } for  pid=2183 comm="mount.nfs" name="utab.IuAaSb" dev="tmpfs" ino=30727 scontext=root:sysadm_r:mount_t:s0-s15:c0.c1023 tcontext=root:object_r:mount_var_run_t:s0 tclass=file
type=AVC msg=audit(1396939321.174:110): avc:  denied  { unlink } for  pid=2183 comm="mount.nfs" name="utab" dev="tmpfs" ino=27419 scontext=root:sysadm_r:mount_t:s0-s15:c0.c1023 tcontext=root:object_r:mount_var_run_t:s0 tclass=file
type=AVC msg=audit(1396936635.289:46): avc:  denied  { getattr } for  pid=2035 comm="mount" name="/" dev="selinuxfs" ino=1 scontext=root:sysadm_r:mount_t:s0-s15:c0.c1023 tcontext=system_u:object_r:security_t:s0 tclass=filesystem

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/userdomain.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/system/userdomain.te b/policy/modules/system/userdomain.te
index d57aaa0..4b57869 100644
--- a/policy/modules/system/userdomain.te
+++ b/policy/modules/system/userdomain.te
@@ -92,6 +92,7 @@ type user_home_t alias { staff_home_t sysadm_home_t secadm_home_t auditadm_home_
 typealias user_home_t alias { staff_untrusted_content_t sysadm_untrusted_content_t secadm_untrusted_content_t auditadm_untrusted_content_t unconfined_untrusted_content_t };
 userdom_user_home_content(user_home_t)
 fs_associate_tmpfs(user_home_t)
+fs_type(user_home_t)
 files_associate_tmp(user_home_t)
 files_poly_parent(user_home_t)
 files_mountpoint(user_home_t)
-- 
2.7.4

